run.py

```diff
@@ -13,78 +16,184 @@
-def close():
+@server.route('/webbrowser/close', methods=['GET', 'POST'])
+@token_required
+def API_close():
     system.close()
-    return "closed\n"
+    return jsonify({'message': 'Webbrowser closed', 'rc': 0})
 
 
-@server.route('/poweroff')
-def poweroff():
+@server.route('/system/poweroff', methods=['GET', 'POST'])
+@token_required
+def API_poweroff():
     system.poweroff()
-    return "poweroff...\n"
+    return jsonify({'message': 'Machine is powering off...', 'rc': 0})
 
 
-@server.route('/reboot')
-def reboot():
+@server.route('/system/reboot', methods=['GET', 'POST'])
+@token_required
+def API_reboot():
     system.reboot()
-    return "reboot...\n"
+    return jsonify({'message': 'Machine is rebooting', 'rc': 0})
 
 
-@server.route('/screenshot')
-def screenshot():
+@server.route('/system/screenshot', methods=['GET', 'POST'])
+@token_required
+def API_screenshot():
     system.screenshot()
-    return "screenshot taken\n"
+    return jsonify({'message': 'Screenshot taken', 'rc': 0})
 
 
-@server.route('/mute')
-def mute():
+@server.route('/system/mute', methods=['GET', 'POST'])
+@token_required
+def API_mute():
     system.mute()
-    return "muted\n"
+    return jsonify({'message': 'Volume muted', 'rc': 0})
+
+
+@server.route('/system/grab_screen', methods=['GET', 'POST'])
+@token_required
+def API_grab_screen():
+    return jsonify({'message': 'screen grabbed',
+                    'rc': 0,
+                    'screen': system.get_screen()})
+
+
+@server.route('/system/set_volume', methods=['GET', 'POST'])
+@token_required
+def API_set_volume():
+    level = flask.request.args.get('lvl')
+    try:
+        level = int(level)
+    except ValueError:
+        return jsonify({'message': f'ERROR: {level} is not int!', 'rc': 1})
+    if not level <= 100 and not level >= 0:
+        return jsonify({'message': f'ERROR: {level} is not in range 0-100', 'rc': 1})
+    system.volume(level)
+    return jsonify({'message': f'Volume set to {level}', 'rc': 0})
+
+
+@server.route('/keyboard/call_key', methods=['GET', 'POST'])
+@token_required
+def API_call_key():
+    key = flask.request.args.get('key')
+    system.xdotool_key(key)
+    return jsonify({'message': 'key called', 'rc': 0})
+
+
+@server.route('/keyboard/call_word', methods=['GET', 'POST'])
+@token_required
+def API_call_word():
+    word = flask.request.args.get('word')
+    system.type_keyboard(word)
+    return jsonify({'message': 'word called', 'rc': 0})
 
 
 @app.callback(
     Output('open-output-message', 'children'),
-    [Input('url-button', 'n_clicks')], [State('url', 'value')])
-def app_open(n_clicks, value):
+    [Input('url-button', 'n_clicks')], [State('url', 'value'),  State('url_history', 'value')])
+def GUI_app_open(n_clicks, value, value_h):
     if n_clicks != 0:
-        system.web_open(value)
+        if value_h is None:
+            system.web_open(value)
+        else:
+            system.web_open(value_h)
     return u'opened'
 
 
+@app.callback(
```

Comment:
As I see this is the only callback that you edit. Why this review touches all the other functions that have nothing to do with this PR?